packages/openai/src/openai.ts

```diff
@@ -174,6 +174,55 @@ export interface OpenAIConfig {
  * - `customizeSpeechParams(params, config)`: Modify speech params
  * - `customizeEmbeddingParams(params, config)`: Modify embedding params
  */
+
+/**
+ * Helper function to convert OpenAI-style usage to the new structured Usage format
+ */
+function convertOpenAIUsage(openaiUsage: {
+  prompt_tokens?: number;
+  completion_tokens?: number;
+  total_tokens?: number;
+  prompt_tokens_details?: { cached_tokens?: number; audio_tokens?: number };
+  completion_tokens_details?: { reasoning_tokens?: number; audio_tokens?: number };
+}): Usage {
+  const usage: Usage = {};
+
+  // Text tokens (default category)
+  const textInput = (openaiUsage.prompt_tokens || 0) - 
+                   (openaiUsage.prompt_tokens_details?.cached_tokens || 0) -
+                   (openaiUsage.prompt_tokens_details?.audio_tokens || 0);
+  const textOutput = (openaiUsage.completion_tokens || 0) - 
+                    (openaiUsage.completion_tokens_details?.reasoning_tokens || 0) -
+                    (openaiUsage.completion_tokens_details?.audio_tokens || 0);
+  
+  if (textInput > 0 || textOutput > 0 || openaiUsage.prompt_tokens_details?.cached_tokens) {
+    usage.text = {};
+    if (textInput > 0) usage.text.input = textInput;
+    if (textOutput > 0) usage.text.output = textOutput;
+    if (openaiUsage.prompt_tokens_details?.cached_tokens) {
+      usage.text.cached = openaiUsage.prompt_tokens_details.cached_tokens;
```

Comment:
Aren't these cached tokens being double counted? Also update pricing calculation in ai.ts if there are cached tokens but but no cache pricing - use input pricing.